### PR TITLE
Add support for logging configuration for TopicOperator provided in CRD

### PR DESCRIPTION
**Pull request by ppietruch**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

Add support for logging configuration for TopicOperator provided in CRD

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging

### CODE REVIEW

**Comment by scholzj** [comment]
Oh, one more think we probably need to solve ... we need to roll the deployment when the configuration changes. Or does Log4j2 reload the configuration from the file _on the fly_? Perhaps you could help with this @stanlyDoge

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TopicOperator.java [hunk 1]
```diff
@@ -30,6 +32,8 @@
 
     private static final String NAME_SUFFIX = "-topic-operator";
 
+    protected static final String METRICS_AND_LOG_CONFIG_SUFFIX = NAME_SUFFIX + "-config";
+
```
[the remark this answers is not in the document]
Comment (ppietruch) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TopicOperator.java]: Done

File: cluster-operator/src/main/resources/topicOperatorDefaultLoggingProperties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+log4j.appender.CONSOLE=org.apache.log4j.ConsoleAppender
```
[the remark this answers is not in the document]
Comment (ppietruch) [reply cluster-operator/src/main/resources/topicOperatorDefaultLoggingProperties]: Done

File: topic-operator/scripts/topic_operator_run.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set +x
+
+if [ -f /opt/topic-operator/custom-config/log4j.properties ];
+then
+    export JAVA_OPTS="${JAVA_OPTS} -Dlog4j.configurationFile=file:/opt/topic-operator/custom-config/log4j.properties"
```
[the remark this answers is not in the document]
Comment (ppietruch) [reply topic-operator/scripts/topic_operator_run.sh]: Done

**Comment by scholzj** [comment]
It looks good to me - I will get tomorrow @stanlyDoge to review it as well as he wrote most of the configuration stuff and merge it afterwards. Thanks

**Comment by scholzj** [comment]
@ppietruch I will fix the last comment while merging the PR.

**Pull request merged**